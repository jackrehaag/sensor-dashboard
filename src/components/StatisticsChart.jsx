import React from "react";
import Chart from "chart.js";

class StatisticsChart extends React.Component {
  state = {};
  chartRef = React.createRef();

  componentDidMount() {
    this.context = this.chartRef.current.getContext("2d");
    this.myChart = new Chart(this.context, {
      type: "line",
      data: {
        xAxisID: "time",
        yAxisID: "value",
        datasets: [
          {
            label: this.props.name,
            data: this.props.data,
            backgroundColor: ["rgba(97, 186, 255, 0.2)"],
            borderColor: ["rgba(97, 186, 255, 1)"],
            borderWidth: 1
          }
        ]
      },
      options: {
        elements: {
          line: {
            tension: 0 // disables bezier curves
          }
        },
        scales: {
          yAxes: [
            {
              ticks: {
                beginAtZero: true
              }
            }
          ],
          xAxes: [
            {
              id: "time",
              type: "time"
            }
          ]
        }
      }
    });
  }

  componentDidUpdate() {
    if (this.props.data) {
      this.myChart.data.datasets[0].data = this.props.data;
      this.myChart.update();
    }
  }

  render() {
    return (
      <div>
        <canvas ref={this.chartRef} width="400" height="270" />
      </div>
    );
  }
}

export default StatisticsChart;
