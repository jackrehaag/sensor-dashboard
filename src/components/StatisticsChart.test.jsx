import React from "react";
import StatisticsChart from "./StatisticsChart";
import renderer from "react-test-renderer";
jest.mock("chart.js");
import Chart from "chart.js";

function createNodeMock(element) {
  if (element.type === "canvas") {
    return {
      getContext() {
        return {};
      }
    };
  }
}

describe("default chart", () => {
  const chartName = "snapshot chart";
  const component = renderer.create(
    <StatisticsChart name={chartName} data={[{ x: 123, y: 456 }]} />,
    {
      createNodeMock
    }
  );
  const tree = component.toJSON();

  test("it renders", () => {
    expect(tree).toMatchSnapshot();
  });

  test("it calls chart.js with the correct config", () => {
    expect(Chart.mock.calls).toMatchSnapshot();
  });
});
