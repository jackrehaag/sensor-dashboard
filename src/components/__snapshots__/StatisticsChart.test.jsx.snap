// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default chart it calls chart.js with the correct config 1`] = `
Array [
  Array [
    Object {},
    Object {
      "data": Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "rgba(97, 186, 255, 0.2)",
            ],
            "borderColor": Array [
              "rgba(97, 186, 255, 1)",
            ],
            "borderWidth": 1,
            "data": Array [
              Object {
                "x": 123,
                "y": 456,
              },
            ],
            "label": "snapshot chart",
          },
        ],
        "xAxisID": "time",
        "yAxisID": "value",
      },
      "options": Object {
        "elements": Object {
          "line": Object {
            "tension": 0,
          },
        },
        "scales": Object {
          "xAxes": Array [
            Object {
              "id": "time",
              "type": "time",
            },
          ],
          "yAxes": Array [
            Object {
              "ticks": Object {
                "beginAtZero": true,
              },
            },
          ],
        },
      },
      "type": "line",
    },
  ],
]
`;

exports[`default chart it renders 1`] = `
<div>
  <canvas
    height="270"
    width="400"
  />
</div>
`;